### DISCUSSION TITLE
v0.3.5

### DISCUSSION THREAD

**Discussion by ilyes319** [ACEsuit/mace#454] [Announcements]
## What's Changed
* on the fly data loading by @davkovacs in https://github.com/ACEsuit/mace/pull/73
* On the fly dataloading by @davkovacs in https://github.com/ACEsuit/mace/pull/81
* Make matscipy default neighbour list by @ilyes319 in https://github.com/ACEsuit/mace/pull/64
* Multi gpu by @ilyes319 in https://github.com/ACEsuit/mace/pull/99
* Slight modification to HDF5Dataset by @sivonxay in https://github.com/ACEsuit/mace/pull/117
* Multi-node, multi-GPU data parallel training. by @samwaltonnorwood in https://github.com/ACEsuit/mace/pull/105
* Multi-processing by @mavaylon1 in https://github.com/ACEsuit/mace/pull/133
* Develop by @ilyes319 in https://github.com/ACEsuit/mace/pull/207
* load_foundations support for max_L other than 2 by @bernstei in https://github.com/ACEsuit/mace/pull/273
* print validation error before any training by @bernstei in https://github.com/ACEsuit/mace/pull/288
* update foundations with develop by @ilyes319 in https://github.com/ACEsuit/mace/pull/299
* Ilyes319 patch 3 by @ilyes319 in https://github.com/ACEsuit/mace/pull/305
* Warning on float64 training with PyTorch 2.1 by @ilyes319 in https://github.com/ACEsuit/mace/pull/306
* Foundations by @ilyes319 in https://github.com/ACEsuit/mace/pull/307
* initial torch.compile support (inference only) by @hatemhelal in https://github.com/ACEsuit/mace/pull/300
* Handle a wider range of possible slurm env vars by @bernstei in https://github.com/ACEsuit/mace/pull/332
* Update plot_train.py by @zhubonan in https://github.com/ACEsuit/mace/pull/352
* add note on mp corrections on energies by @chiang-yuan in https://github.com/ACEsuit/mace/pull/342
* Update compile test case to use larger test system by @hatemhelal in https://github.com/ACEsuit/mace/pull/310
* Adding pre-commit checks and github action runners by @hatemhelal in https://github.com/ACEsuit/mace/pull/302
* Add torch.compile compatiblity by @ilyes319 in https://github.com/ACEsuit/mace/pull/354
* Add an option to set the D3 damping function, which defaults to D3(BJ) by @Andrew-S-Rosen in https://github.com/ACEsuit/mace/pull/268
* Use HTTPS for MACE MP checkpoint URLs by @janosh in https://github.com/ACEsuit/mace/pull/318
* Add multi-GPU, data parallelisation, pair repulsion and distance transforms. by @ilyes319 in https://github.com/ACEsuit/mace/pull/359
* Add multi-GPU dataparallel by @ilyes319 in https://github.com/ACEsuit/mace/pull/74
* Add mutli-GPU training, data parrallelisation, pair repulsion, distance transform and other features by @ilyes319 in https://github.com/ACEsuit/mace/pull/363
* Add metadata saving to compiled model and hot fix stress extraction by @ilyes319 in https://github.com/ACEsuit/mace/pull/387
* Fix workflows check by @ilyes319 in https://github.com/ACEsuit/mace/pull/388
* add GitPython to deps by @ilyes319 in https://github.com/ACEsuit/mace/pull/389
* add pylint to pre-commit by @ilyes319 in https://github.com/ACEsuit/mace/pull/391
* fixed readme typos, and commit by @ilyes319 in https://github.com/ACEsuit/mace/pull/399
* make cli scripts importable from python by @alinelena in https://github.com/ACEsuit/mace/pull/398
* Develop by @ilyes319 in https://github.com/ACEsuit/mace/pull/418
* Improve warning for "energy", "forces" key by @ilyes319 in https://github.com/ACEsuit/mace/pull/419
* log errors and handle checkpoint io on rank 0 only by @jharrymoore in https://github.com/ACEsuit/mace/pull/420
* Fix checkpoint for multi-GPU by @ilyes319 in https://github.com/ACEsuit/mace/pull/421
* removed wront argument parsing in preprocess_data by @VondrakMar in https://github.com/ACEsuit/mace/pull/423
* Fix preprocess data and pylint by @ilyes319 in https://github.com/ACEsuit/mace/pull/424
* fix atom e0s being subtracted twice in mace calculator by @RokasEl in https://github.com/ACEsuit/mace/pull/425
* fix atom e0s being subtracted twice in mace calculator by @ilyes319 in https://github.com/ACEsuit/mace/pull/426
* fix bug test set stress key by @ilyes319 in https://github.com/ACEsuit/mace/pull/428

## New Contributors
* @samwaltonnorwood made their first contribution in https://github.com/ACEsuit/mace/pull/105
* @mavaylon1 made their first contribution in https://github.com/ACEsuit/mace/pull/133
* @hatemhelal made their first contribution in https://github.com/ACEsuit/mace/pull/300
* @zhubonan made their first contribution in https://github.com/ACEsuit/mace/pull/352
* @Andrew-S-Rosen made their first contribution in https://github.com/ACEsuit/mace/pull/268
* @alinelena made their first contribution in https://github.com/ACEsuit/mace/pull/398
* @VondrakMar made their first contribution in https://github.com/ACEsuit/mace/pull/423
* @RokasEl made their first contribution in https://github.com/ACEsuit/mace/pull/425

**Full Changelog**: https://github.com/ACEsuit/mace/compare/v0.3.4...v0.3.5

<hr /><em>This discussion was created from the release <a href='https://github.com/ACEsuit/mace/releases/tag/v0.3.5'>v0.3.5</a>.</em>
